cpmpy/solvers/pysat.py

```diff
@@ -465,9 +465,10 @@ def _pysat_cardinality(self, cpm_expr, reified=False):
             raise ValueError(f"PySAT: Expected Comparison to be either <=, ==, or >=, but was {cpm_expr.name}")
 
     def _pysat_pseudoboolean(self, cpm_expr):
-        """ Convert CPMpy comparison of `wsum` (over Boolean variables) into PySAT list of clauses """
-        if self._pb is None:
-            raise self.IMPORT_PYSAT_PBLIB_ERROR
+        """Convert CPMpy comparison of `wsum` (over Boolean variables) into PySAT list of clauses."""
+        if self._pb is False:
+            raise ImportError("The model contains a PB constraint, for which PySAT needs an additional dependency (PBLib). To install it, run `pip install pypblib`.")
+        assert self._pb is not None, "We should have imported the `pb` module if it was installed."
```

Comment:
now I'm just confused... you want it to be eitehr false or none?
Just one is fine..,

Reply:
I've added a doc string on `_pb` to clarify, but there are (confusingly!) three states, where `None` is we didn't import yet, the module where we did import succesfully and `False` where we tried but it was not imported (so not installed). This *really* prevents us from attempting imports multiple times whether it is installed or not.

You can let me know if you think is ok. Alternative design was to indeed omit the field (as "no import yet"). I thought I'd then have to use `has_attr(CPM_py, "_pb")` or something to distinguish.. but actually this might be more simple and clear, so I will try that now.

Reply:
Well, I feel this is a little more clear, let me know. Obviously, all quite overkill for something that is only checked when we initialize a solver, which makes code readability more of a priority.

Reply:
More generally, we do a lot of engineering to avoid import statements, because (so I heard) they cause significant overhead. When we have more regular benchmarking, we should test if this is still true, or is also true if we use [`importlib.find_spec`](https://docs.python.org/3/library/importlib.html#importlib.util.find_spec).